packages/components/overflow-items/overflow-items.ts

```diff
@@ -0,0 +1,291 @@
+            </div>
+        }
+
+        @if (!reverseOverflowOrder()) {
+            <ng-container [ngTemplateOutlet]="template" />
+        }
+
+        <ng-template #template>
+            <div
+                class="kbq-overflow-items__result"
+                #resultRef
+                [attr.aria-hidden]="resultHidden()"
+                [class.kbq-overflow-items__result_hidden]="resultHidden()"
+            >
+                <ng-container
+                    [ngTemplateOutlet]="result()?.templateRef || null"
+                    [ngTemplateOutletContext]="resultTemplateContext()"
+                />
+            </div>
+        </ng-template>
+    `,
+    styleUrl: './overflow-items.scss',
+    host: {
+        class: 'kbq-overflow-items'
+    },
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class KbqOverflowItems {
+    private readonly itemElementRefs = viewChildren<string, ElementRef<HTMLElement>>('itemRef', { read: ElementRef });
+
+    /**
+     * `KbqOverflowItem` directive references.
+     *
+     * @docs-private
+     */
+    protected readonly items = contentChildren(KbqOverflowItem);
+
+    /**
+     * `KbqOverflowItemsResult` directive reference.
+     *
+     * @docs-private
+     */
+    protected readonly result = contentChild(KbqOverflowItemsResult);
+
+    private readonly resultElementRef = viewChild<string, ElementRef<HTMLElement>>('resultRef', { read: ElementRef });
+
+    /**
+     * Template context for the `KbqOverflowItemsResult` directive.
+     *
+     * @docs-private
+     */
+    protected readonly resultTemplateContext = computed<KbqOverflowItemsResultContext>(() => {
+        const hiddenItemIDs = new Set(
+            this.items()
+                .filter(({ hidden }) => hidden())
+                .map(({ id }) => id())
+        );
+        return {
+            $implicit: hiddenItemIDs,
+            hiddenItemIDs: hiddenItemIDs
+        };
+    });
+
+    /**
+     * Whether the overflow result is hidden.
+     *
+     * @docs-private
+     */
+    protected readonly resultHidden = computed(() => !this.items().some(({ hidden }) => hidden()));
+
+    /**
+     * Whether the overflow order should be reversed.
+     * Also changes the position of the `resultTemplateRef`.
+     *
+     * @default false
+     */
+    readonly reverseOverflowOrder = input(false, { transform: booleanAttribute });
+
+    private readonly elementRef = inject<ElementRef<HTMLElement>>(ElementRef);
+    private readonly resizeObserver = inject(SharedResizeObserver);
+    private readonly contentObserver = inject(ContentObserver);
+
+    constructor() {
+        toObservable(this.reverseOverflowOrder)
+            .pipe(skip(1), takeUntilDestroyed())
+            .subscribe(() => {
+                this.items().forEach(({ hidden }) => hidden.set(false));
+            });
+
+        merge(
+            toObservable(this.items),
+            this.resizeObserver.observe(this.elementRef.nativeElement),
```

Comment:
Давай сразу заложим проверку, потому что видел, что у ребят, которые на SSR пересаживаются, начали баги возникать

```suggestion
            this.resizeObserver?.observe(this.elementRef.nativeElement),
```

Reply:
здесь используется [SharedResizeObserver](https://github.com/angular/components/blob/cfc8cf7ebd36b0579bea87626ec09646db8b8f83/src/cdk/observers/private/shared-resize-observer.ts#L100), он всегда доступен

у ребят была проблема из-за использования window в коде компонента

Reply:
[проверка уже есть](https://github.com/angular/components/blob/cfc8cf7ebd36b0579bea87626ec09646db8b8f83/src/cdk/observers/private/shared-resize-observer.ts#L108)

Reply:
кайф! надо будет его затянуть в другие места потом тоже